rir/src/interpreter/cache.h

```diff
@@ -0,0 +1,125 @@
+#ifndef RIR_INTERPRETER_CACHE_H
+#define RIR_INTERPRETER_CACHE_H
+
+#include "instance.h"
+
+namespace rir {
+
+// must be a power of 2 for modulus using & CACHE_MASK to work
+#define MAX_CACHE_SIZE 128
+#define CACHE_MASK (MAX_CACHE_SIZE - 1)
```

Comment:
ok, but now I really don't see why this is still needed. just don't create more than MAX_CACHE_SIZE worth of cached loads/stores

Reply:
What if a function has more bindings? Is there a precise limit?

Reply:
as I said, just don't cache more. you can decide to emit cached/non-cached loads. if you have say more than 200 env/symbol pairs, then just stop emiting cached ones. this automatically ensures that the index is never bigger than 200.

Reply:
@charig the immediateIdx should not be needed anymore, that was my point for this change...

Reply:
As I've been saying, we still use the idx to distinguish from uninitialized. I do not understand precisely what do you have in mind for optimizing here.